Libplanet.Extensions.Cocona/Commands/StoreQueryCommand.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Immutable;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Bencodex.Types;
+using Cocona;
+using Libplanet.Action;
+using Libplanet.Blocks;
+using Libplanet.Tx;
+
+namespace Libplanet.Extensions.Cocona.Commands
+{
+    public class StoreQueryCommand
+    {
+        [Command(Description = "Query block by the index")]
+        public void BlockByIndex(
+            [Argument("HOME", Description = "root home path")]
+            string home,
+            [Argument("BLOCK-INDEX", Description = "block index")]
+            int blockIndex
+        )
+        {
+            var store = new RocksDBStore.RocksDBStore(home);
+            if (!(store.GetCanonicalChainId() is { } chainId))
+            {
+                throw Utils.Error("Cannot find the main branch of the blockchain.");
+            }
+
+            if (!(store.IndexBlockHash(chainId, blockIndex) is { } blockHash))
+            {
+                throw Utils.Error(
+                    $"Cannot find the block with the height {blockIndex}" +
+                    $" within the blockchain {chainId}."
+                );
+            }
+
+            if (!(store.GetBlock<DummyAction>(blockHash) is { } block))
+            {
+                throw Utils.Error($"Cannot find the block with such hash: {blockHash}.");
+            }
+
+            Console.WriteLine(SerializeHumanReadable(block));
+        }
+
+        [Command(Description = "Query block by the hash")]
+        public void BlockByHash(
+            [Argument("HOME", Description = "root home path")]
+            string home,
+            [Argument("BLOCK-HASH", Description = "block hash")]
+            string blockHash
+        )
+        {
+            var store = new RocksDBStore.RocksDBStore(home);
```

Comment:
As there are multiple `IStore` implementations, we need to be more flexible on this.  The way `MptCommand` deals with multiple `IKeyValueStore` implementations can be applied here too.

<https://github.com/planetarium/libplanet/blob/77a6eb6d175f0ed550bbf01b1fe0afc17e625887/Libplanet.Extensions.Cocona/Commands/MptCommand.cs#L243-L272>

Reply:
Applied. Thanks